Lib/glyphsLib/classes.py

```diff
@@ -1959,6 +1944,45 @@ def __repr__(self):
         else:
             return "<GSHint %s %s>" % (self.type, direction)
 
+    @property
+    def originNode(self):
+        if self._originNode is not None:
+            return self._originNode
+        if self._origin is not None:
+            # Find the GSNode that is refered to by the indices in _origin
+            path_index, node_index = self._origin
+            layer = self.parent  # FIXME: (jany) I don't have access to the parent from the GSHint!!!
+            path = layer.paths[path_index]
+            node = path.nodes[node_index]
+            return node
+
+    @originNode.setter
+    def originNode(self, node):
+        self._originNode = node
+        self._origin = None
+
+    @property
+    def origin(self):
+        if self._origin is not None:
+            return self._origin
+        if self._originNode is not None:
+            # Find the path_index & node_index of the _originNode
+            path = self._originNode.parent
+            layer = path.parent
+            for path_index in range(len(layer.paths)):
+                if path == layer.paths[path_index]:
+                    for node_index in range(len(path.nodes)):
+                        if self._originNode == path.nodes[node_index]:
+                            return point(path_index, node_index)
+            return None
+
+    @origin.setter
+    def origin(self, origin):
+        self._origin = origin
+        self._originNode = None
+
+    # FIXME: (jany) if the above is OK after review, do the same for the others
```

Comment:
@moyogo Please review this

Reply:
We can drop `hint.origin` as the GlyphsSDK has `hint.originNode` instead. The same goes for  and `hint.targetNode`, `hint.otherNode1` and `hint.otherNode2`. I think `hint.origin` is an objc thing, not part of the Python API.

We can also pythonize the names of `Hint__origin__pos()` to `_origin_pos()` and `Hint__width__pos()` to `_width_pos()`.